### ISSUE TITLE
Hopfield different activation

### ISSUE THREAD

**Issue by zorantodorovic** [wagenaartje/neataptic#20]
Is it possible to create Hopfield with some other activation function, other than STEP?
Also, do you have some ideas how to implement self-organizing map?

**Comment by wagenaartje** [comment]
Hi, why would you want to create the hopfield network without the STEP function? It basically rounds the output to `0` or `1`. Having a Hopfield network with for example the Sigmoid function would be the same as using `new Network(size, size)`. 

Self-organizing map? Could you be more specific? Do you mean like a route between two points on a map? Glad to give some ideas.

**Comment by zorantodorovic** [comment]
@wagenaartje you're right, STEP function is sufficient. Your example is great. Is it possible to set learning rate to Hopfield network?
SOM reduces data dimensions and displays similarities among data.

**Comment by wagenaartje** [comment]
Ofcourse you can set the learning rate, example:

```javascript
var network = Architect.Hopfield(10);
var trainingSet = [
  { input: [0, 1, 0, 1, 0, 1, 0, 1, 0, 1], output: [0, 1, 0, 1, 0, 1, 0, 1, 0, 1] },
  { input: [1, 1, 1, 1, 1, 0, 0, 0, 0, 0], output: [1, 1, 1, 1, 1, 0, 0, 0, 0, 0] }
];

// Learning rate of 0.01
network.train(trainingSet, {
  rate: 0.1,
  log: 1
});

network.activate([0,1,0,1,0,1,0,1,1,1]); // [0, 1, 0, 1, 0, 1, 0, 1, 0, 1]
network.activate([1,1,1,1,1,0,0,1,0,0]); // [1, 1, 1, 1, 1, 0, 0, 0, 0, 0]
```

[SOM](https://en.wikipedia.org/wiki/Self-organizing_map) looks really interesting. However it is very different from the neural networks that Neataptic provides, especially the learning algorithm.

![](https://upload.wikimedia.org/wikipedia/commons/thumb/7/70/Synapse_Self-Organizing_Map.png/300px-Synapse_Self-Organizing_Map.png)

I might start developing this somewhere, but i've got a lot to do first!

**Comment by wagenaartje** [comment]
I'm closing this for now.
